### DISCUSSION TITLE
How I can stop search

### DISCUSSION THREAD

**Discussion by vuphamngoctan1612** [google/or-tools#4437] [CP-SAT questions] [answered]
I have problem and i use minimize to solve it. But i want to stop search when objective value = or < 3. What should i do?

**Comment by sschnug** [comment, marked as answer]
See;

- [StopAfterNSolutionsSampleSat.cs](https://github.com/google/or-tools/blob/main/ortools/sat/samples/StopAfterNSolutionsSampleSat.cs)
- [StopAfterNSolutionsSampleSat.java](https://github.com/google/or-tools/blob/main/ortools/sat/samples/StopAfterNSolutionsSampleSat.java)
- [stop_after_n_solutions_sample_sat.cc](https://github.com/google/or-tools/blob/main/ortools/sat/samples/stop_after_n_solutions_sample_sat.cc)
- [stop_after_n_solutions_sample_sat.py](https://github.com/google/or-tools/blob/main/ortools/sat/samples/stop_after_n_solutions_sample_sat.py)

(slightly different styles in above examples!)

While in those examples, the **limit** is an upper-bound on some (addition-operator) accumulation (one event = new incumbent), in your case the **limit** is an lower-bound on some (min-operator) accumulation (on incumbent values): it should be easy to adapt!
